configuration/device-tree.md

```diff
@@ -448,9 +448,60 @@ When assigning to the `reg` property, the address portion of the parent node nam
 
 The `name` property is a pseudo-property - it shouldn't appear in a DT, but assigning to it causes the name of its parent node to be changed to the assigned value. Like the `reg` property, this can be used to give nodes unique names.
 
-
 <a name="part2.2.10"></a>
-#### 2.2.10 Examples
+#### 2.2.10 The overlay map file
+
+The introduction of the Pi 4, built around the BCM2711 SoC, brought with it many changes; some of these changes are additional interfaces, and some are modifications to (or removals of) existing interfaces. There are new overlays intended specifically for the Pi 4 that don't make sense on older hardware, e.g. overlays that enable the new SPI, I2C and UART interfaces, but other overlays don't apply correctly even though they control features that are still relevant on the new device.
+
+There is therefore a need for a method of tailoring an overlay to multiple platforms with differing hardware. Supporting them all in a single .dtbo file would require heavy use of hidden ("dormant") fragments and a switch to an on-demand symbol resolution mechanism so that a missing symbol that isn't needed doesn't cause a failure. A simpler solution is to add a facility to map an overlay name to one of several implementation files depending on the current platform.
+
+The overlay map, which is rolling out with the switch to Linux 5.4, is a file that gets loaded by the firmware at start of day. It is written in DTS source format - `overlay_map.dts`, compiled to `overlay_map.dtb` and stored in the overlays directory.
+
+This is an edited version of the current map file (the full version is available [here](https://github.com/raspberrypi/linux/blob/rpi-5.4.y/arch/arm/boot/dts/overlays/overlay_map.dts)):
+
+```
+/ {
+    vc4-kms-v3d {
+        bcm2835;
+        bcm2711 = "vc4-kms-v3d-pi4";
+    };
+
+    vc4-kms-v3d-pi4 {
+        bcm2711;
+    };
+
+    uart5 {
+        bcm2711;
+    };
+
+    pi3-disable-bt {
+        renamed = "disable-bt";
+    };
+
+    lirc-rpi {
+        deprecated = "use gpio-ir";
+    };
+};
+```
+
+Each node has the name of an overlay that requires special handling. The properties of each node are either platform names or one of a small number of special directives. A platform name with no value (an empty property) indicates that the current overlay is compatible with the platform; for example, `vc4-kms-v3d` is compatible with the `bcm2835` platform. A non-empty value for a platform is the name of an alternative overlay to use in place of the requested one; asking for `vc4-kms-v3d` on BCM2711 results in `vc4-kms-v3d-pi4` being loaded instead. Any platform not included in an overlay's node is not compatible with that overlay.
```

Comment:
"compatible with the `bcm2835` platform."

Is it worth clarifying that this also includes the BCM2836 and BCM2837 SoCs? (or is that something that the reader of this page will already know?)

Reply:
It's mentioned in section 2.1 (Fragments):

> The `compatible` string identifies this as being for BCM2835, which is the base architecture for the Raspberry Pi SoCs; if the overlay makes use of features of a Pi 4 then `brcm,bcm2711` is the correct value to use, otherwise `brcm,bcm2835` can be used for all Pi overlays.